packages/rpc-methods/src/restricted/manageAccounts.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  SubjectType,
+  PermissionType,
+  RestrictedMethodOptions,
+  ValidPermissionSpecification,
+  PermissionSpecificationBuilder,
+} from '@metamask/permission-controller';
+import { Json, NonEmptyArray } from '@metamask/utils';
+import { ethErrors } from 'eth-rpc-errors';
+
+export const methodName = 'snap_manageAccounts';
+
+export type ManageAccountsMethodHooks = {
+  /**
+   * Gets the snap keyring implementation.
+   */
+  getSnapKeyring: (snapOrigin: string) => Promise<{
+    handleKeyringSnapMessage: (
+      snapId: string,
+      message: any,
+      // eslint-disable-next-line @typescript-eslint/ban-types
+      saveSnapKeyring: Function,
+    ) => Promise<Json>;
+  }>;
+
+  /**
+   * Saves the snap keyring, should be called after mutable operations.
+   *
+   * If an account was removed the address of the account must be passed to
+   * update the UI.
+   */
+  saveSnapKeyring: (removedAddress?: string) => Promise<void>;
+};
+
+type ManageAccountsSpecificationBuilderOptions = {
+  allowedCaveats?: Readonly<NonEmptyArray<string>> | null;
+  methodHooks: ManageAccountsMethodHooks;
+};
+
+type ManageAccountsSpecification = ValidPermissionSpecification<{
+  permissionType: PermissionType.RestrictedMethod;
+  targetName: typeof methodName;
+  methodImplementation: ReturnType<typeof manageAccountsImplementation>;
+  allowedCaveats: Readonly<NonEmptyArray<string>> | null;
+}>;
+
+/**
+ * The specification builder for the `snap_manageAccounts` permission.
+ * `snap_manageAccounts` lets the Snap manage a set of accounts via a custom keyring.
+ *
+ * @param options - The specification builder options.
+ * @param options.allowedCaveats - The optional allowed caveats for the permission.
+ * @param options.methodHooks - The RPC method hooks needed by the method implementation.
+ * @returns The specification for the `snap_manageAccounts` permission.
+ */
+export const specificationBuilder: PermissionSpecificationBuilder<
+  PermissionType.RestrictedMethod,
+  ManageAccountsSpecificationBuilderOptions,
+  ManageAccountsSpecification
+> = ({
+  allowedCaveats = null,
+  methodHooks,
+}: ManageAccountsSpecificationBuilderOptions) => {
+  return {
+    permissionType: PermissionType.RestrictedMethod,
+    targetName: methodName,
+    allowedCaveats,
+    methodImplementation: manageAccountsImplementation(methodHooks),
+    subjectTypes: [SubjectType.Internal, SubjectType.Snap],
```

Comment:
Why does this need to be available outside of snaps?

Reply:
This is also not valid anymore, I'll update the code

Reply:
Updated in https://github.com/MetaMask/snaps/pull/1290/commits/7bb264c1f9593b9c1ab4a1c24e0d0fe6eab40c9a